Inorder timeouts

Jazzepi opened the pull request:
*Fixes issue https://code.google.com/archive/p/mockito/issues/292
  *Updated code provided by https://github.com/sirchia in PR #20

Closes #20

ChristianSchwarz commented on src/main/java/org/mockito/verification/VerificationWrapperInOrderWrapper.java:
```diff
@@ -0,0 +1,49 @@
+        if (verificationMode instanceof VerificationInOrderMode) {
+            final VerificationInOrderMode verificationInOrderMode = (VerificationInOrderMode)verificationMode;
+            return new InOrderWrapper(verificationInOrderMode, inOrder);
+        } else if (verificationMode instanceof VerificationOverTimeImpl) {
```
@Jazzepi to improve readability you can remove the else-keyword here

Jazzepi replied:
Done

Jazzepi replied:
Done

ChristianSchwarz commented on src/main/java/org/mockito/verification/VerificationWrapperInOrderWrapper.java:
```diff
@@ -0,0 +1,49 @@
+        } else if (verificationMode instanceof VerificationOverTimeImpl) {
+            final VerificationOverTimeImpl verificationOverTime = (VerificationOverTimeImpl)verificationMode;
+            if (verificationOverTime.isReturnOnSuccess()) {
+                return new VerificationOverTimeImpl(verificationOverTime.getPollingPeriodMillis(),
```
@Jazzepi The readability should be improved here.You should pass local variables as parameter to the constructor. You could also consider to move the whole construction process into a new method because of its complexity e.g. `wrapVerificationOverTime(..)` .

Jazzepi replied:
I'm not sure what you're asking for in this case. Are you looking for the parameters to be assigned to a local variable before being passed into new InOrderWrapper? Like this?

```
private VerificationMode wrapInOrder(VerificationWrapper<?> verificationWrapper, VerificationMode verificationMode, InOrderImpl inOrder) {
    VerificationMode myVerificationMode = verificationMode;
    InOrderImpl myInOrder = inOrder;
    if (verificationMode instanceof VerificationInOrderMode) {
        final VerificationInOrderMode verificationInOrderMode = (VerificationInOrderMode)verificationMode;
        return new InOrderWrapper(verificationInOrderMode, inOrder);
    }
}
```

ChristianSchwarz replied:
What I wanted to say is that the parameters passed to the constructor should be assigned to local variables first. This avoids interleaved calls for the construction and improves clarity.

```
private static VerificationOverTime wrapVerificationOverTime(VerificationOverTime verificationOverTime, VerificationWrapper verificationWrapper, InOrder inOrder){
    long pollPeriodeMillis =verificationOverTime.getPollingPeriodMillis();
    long durationMillis = verificationOverTime.getTimer().duration();
    InOrder warppedInOrder=wrapInOrder(verificationWrapper, verificationOverTime.getDelegate(), inOrder);
    boolean returnOnSuccess= verificationOverTime.isReturnOnSuccess();

    return new VerificationOverTimeImpl(pollPeriodeMillis, durationMillis, wrappedInOrder, returnOnSuccess);
}
```

TimvdLippe commented:
Could you please add tests to make sure the code is correctly behaving?
Also you have some unrelated changes, mostly imports reordering. Please remove those as well.

Jazzepi commented:
I should have some time to work on the code changes this weekend.

Jazzepi commented:
@TimvdLippe There was already a test there, I just removed the @Ignore tag. I'll add one for the exception throwing behavior as well. I also removed the import reordering. My IDE was trying to be super helpful!

TimvdLippe commented:
@Jazzepi Sounds great! :+1: 

The codecov status has to pass before this can be merged. After the current feedback has been processed I will take another look, but I think so far it is looking good!

ChristianSchwarz commented on src/main/java/org/mockito/verification/Timeout.java:
```diff
@@ -63,4 +66,9 @@ public VerificationMode never() {
+
+    @Override
+    public void verifyInOrder(VerificationDataInOrder data) {
+        System.out.println(data);
```
That one should be removed. Can you explain why this method is empty?

Jazzepi replied:
That was an accident. I've removed it in the latest commit. Haven't delivered it yet. This was leftover with testing to see if I could get the refactoring working without doing any wrapping class weirdness.

TimvdLippe commented:
@Jazzepi could you rebase this PR and commit your changes? I would like to merge this sooner rather than later.

Jazzepi commented:
@TimvdLippe I think it should all be in working ordering now. I've been doing git force pushes to my branch. No idea if that's the accepted style, but Github seems to be figuring out what I'm doing. I just wish it were layering patchsets on like Gerrit does.
@ChristianSchwarz I really puzzled over what you were looking for and could not understand how I could improve the code in the way you're describing. I'd prefer this changeset get merged and then make whatever style changes you wanted yourself.

TimvdLippe commented:
LGTM thanks!
